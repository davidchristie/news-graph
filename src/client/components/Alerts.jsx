import React, { PropTypes } from 'react'

import Alert from './Alert'

export default class Alerts extends React.Component {
  render () {
    return (
      <div>
        {this.props.alerts.map(({text, type}, index) => <Alert key={index} text={text} type={type} />)}
      </div>
    )
  }
}

Alerts.propTypes = {
  alerts: PropTypes.arrayOf(PropTypes.shape({
    text: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired
  })).isRequired
}
