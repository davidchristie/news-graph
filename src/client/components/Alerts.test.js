/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Alerts from './Alerts'

const alerts = [
  {
    text: 'Text',
    type: 'info'
  }
]

test('Alerts renders correctly', () => {
  const wrapper = shallow(<Alerts alerts={alerts} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
