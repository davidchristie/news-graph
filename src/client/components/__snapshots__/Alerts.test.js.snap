exports[`test Alerts renders correctly 1`] = `
<div>
  <Alert
    text="Text"
    type="info" />
</div>
`;
